### PR TITLE
Add: windowshade for rail station construction window

### PR DESCRIPTION
**Pull request by andythenorth**
Rationale for this: 

* I run OpenTTD at 2x UI zoom in a 13" screen
* I use newgrf stations
* The station construction window *always* fills most of the viewport, obscuring the map where I want to build
* I have to drag the station window, which puts it mostly out of the viewport, then drag it back if I want to change station tile or orientation
* windowshade is established in OpenTTD and is more convenient than dragging the window around 

There is a movie of the problem here: https://dev.openttdcoop.org/attachments/download/8265/building_stations.mov (76MB, H264, but .mov format sorry).

### CODE REVIEW

**Comment by andythenorth** [comment]
shade_rail_station_selection

File: src/rail_gui.cpp [hunk 1]
```diff
@@ -1029,21 +1029,23 @@ struct BuildRailStationWindow : public PickerWindowBase {
...
+            top = DrawStationCoverageAreaText(left, right, top, SCT_ALL, rad, false) + WD_PAR_VSEP_NORMAL;
+            top = DrawStationCoverageAreaText(left, right, top, SCT_ALL, rad, true) + WD_PAR_VSEP_NORMAL;
+            /* Resize background if the window is too small.
+             * Never make the window smaller to avoid oscillating if the size change affects the acceptance.
+             * (This is the case, if making the window bigger moves the mouse into the window.) */
+            if (top > bottom) {
+                this->coverage_height += top - bottom;
+                this->ReInit();
+            }
+        }
```
Comment (PeterN) [inline src/rail_gui.cpp]: Wrong indentation, use tabs not spaces.

File: src/rail_gui.cpp [hunk 2]
```diff
@@ -1029,21 +1029,23 @@ struct BuildRailStationWindow : public PickerWindowBase {
...
-		top = DrawStationCoverageAreaText(left, right, top, SCT_ALL, rad, false) + WD_PAR_VSEP_NORMAL;
-		top = DrawStationCoverageAreaText(left, right, top, SCT_ALL, rad, true) + WD_PAR_VSEP_NORMAL;
-		/* Resize background if the window is too small.
-		 * Never make the window smaller to avoid oscillating if the size change affects the acceptance.
-		 * (This is the case, if making the window bigger moves the mouse into the window.) */
-		if (top > bottom) {
-			this->coverage_height += top - bottom;
-			this->ReInit();
-		}
+        if (!this->IsShaded()) {
```
Comment (PeterN) [inline src/rail_gui.cpp]: Change this to an early return and you don't need to re-indent the below.

File: bin/ai/regression/tst_regression/info.nut [hunk 3]
```diff
@@ -0,0 +1,15 @@
+/* $Id$ */
```
Comment (PeterN) [inline bin/ai/regression/tst_regression/info.nut]: Seems unrelated

**Comment by michicc** [comment]
`Add: Windowshade...`

`Change:` should be the last resort commit message tag, no the first choice like it seems to become.

**Review by PeterN — APPROVED** [review]
